services/frontend/pkg/controller/controller.go

```diff
@@ -0,0 +1,269 @@
+				currentState.State = v1beta.STATUS_DONE_ERROR
+			}
+		}
+	} else {
+		currentState.State = v1beta.STATUS_DONE
+	}
+
+	// DELETE && FINALIZER LOGIC
+	// Check if resource isn't schedguled for deletion,
+	// when true, add finalizer else run delete logic
+	if inputManifest.GetDeletionTimestamp().IsZero() {
+		if !controllerutil.ContainsFinalizer(inputManifest, finalizerName) {
+			controllerutil.AddFinalizer(inputManifest, finalizerName)
+			if err := r.kc.Update(ctx, inputManifest); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed executing finalizer: %w", err)
+			}
+		}
+	} else {
+
+		if !configExists {
+			log.Info("Config has been destroyed. Removing finalizer.", "status", currentState.State)
+			controllerutil.RemoveFinalizer(inputManifest, finalizerName)
+			if err := r.kc.Update(ctx, inputManifest); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed removing finalizer: %w", err)
+			}
+			return ctrl.Result{}, nil
+		}
+
+		if !configInDesiredState {
+			inputManifest.SetUpdateResourceStatus(currentState)
+			if err := r.kc.Status().Update(ctx, inputManifest); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed updating status: %w", err)
+			}
+			log.Info("Refreshing state", "status", currentState.State)
+			return ctrl.Result{RequeueAfter: REQUEUE_IN_PROGRES}, nil
+		}
+
+		if controllerutil.ContainsFinalizer(inputManifest, finalizerName) {
+			// Prevent calling deleteConfig, when the deleteConfig call
+			// won't make it yet to scheduler
+			if inputManifest.Status.State == v1beta.STATUS_SCHEDULED_FOR_DELETION {
+				return ctrl.Result{RequeueAfter: REQUEUE_NEW}, nil
+			}	
+			// schedgule deletion of manifest
+			inputManifest.SetDeletingStatus()
+			if err := r.kc.Status().Update(ctx, inputManifest); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed updating status: %w", err)
+			}
+			log.Info("Calling delete config")
+			if err := r.deleteConfig(&rawManifest); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: REQUEUE_DELETE}, nil
+		}
+		return ctrl.Result{RequeueAfter: REQUEUE_DELETE}, nil
+	}
+
+	// Skip the inputManifests thath are ready
+	if !configInDesiredState {
+		// CREATE LOGIC
+		// Call create config if it not present in DB
+		if !configExists {
+			// Prevent calling createConfig, when the inputManifest
+			// won't make it yet to DB
+			if inputManifest.Status.State == v1beta.STATUS_NEW {
+				return ctrl.Result{RequeueAfter: REQUEUE_NEW}, nil
+			}
+			inputManifest.SetNewReousrceStatus()
+			if err := r.kc.Status().Update(ctx, inputManifest); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed executing finalizer: %w", err)
+			}
+			log.Info("Calling create config")
+			if err := r.createConfig(&rawManifest); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: REQUEUE_NEW}, nil
+		}
+
+		// PROVISIONING LOGIC
+		// Refresh inputManifest.status fields
+		if !configInDesiredState {
```

Comment:
The outer `if` already checks for this condition, isn't this if unnecessary ?

Reply:
It isn't - thank you 👍